compiler/tools/fpp-to-json/test/python/locationMapValidator.py

```diff
@@ -0,0 +1,50 @@
+import json
+import sys
+import re
+
+def collect_ids(json_data):
+    ids = []
+
+    if isinstance(json_data, list):
+        for item in json_data:
+            ids.extend(collect_ids(item))
+
+    elif isinstance(json_data, dict):
+        for key, value in json_data.items():
+            if key == 'id':
+                # Check if the value is an integer
+                if isinstance(value, int):
+                    ids.append(value)
+                # Check if the value is a string and matches the format 'id : number'
+                elif isinstance(value, str) and re.match(r"id\s*:\s*\d+$", value):
+                    ids.append(int(value.split(':')[1].strip()))
+
+            if isinstance(value, (dict, list)):
+                ids.extend(collect_ids(value))
+
+    return ids
+
+
+def check_if_ids_in_loc_map(numbers, data):
+    
+    
+    for number in numbers:
+        if str(number) not in data:
+            return f"ID in ast not found in location map: {number}"
+    
+    return True
+
+
+
+with open('fpp-ast.json') as json_file:
```

Comment:
I think we can put the two open constructs together and have just one level of indenting. See https://stackoverflow.com/questions/26731720/how-to-avoid-nested-with-statements-when-working-with-multiple-files-in-python.